common/src/main/java/com/sequenceiq/cloudbreak/util/DatabaseCommon.java

```diff
@@ -1,85 +1,129 @@
-        Matcher matcher = DATABASE_TYPE.matcher(connectionURL);
-        return matcher.matches() ? Optional.of(matcher.group(1)) : Optional.empty();
+    public static String getJdbcConnectionUrl(String vendorDriverId, String host, int port, Optional<String> database) {
+        return getJdbcConnectionUrl(new JdbcConnectionUrlFields(vendorDriverId, host, port, database));
     }
 
-    public static String getConnectionURL(String vendorDriverId, String host, int port, Optional<String> database) {
+    public static String getJdbcConnectionUrl(JdbcConnectionUrlFields fields) {
         String url;
-        switch (vendorDriverId) {
+        switch (fields.getVendorDriverId()) {
             case "postgresql":
-                url = String.format("jdbc:postgresql://%s:%d/", host, port);
-                if (database.isPresent()) {
-                    url += database.get();
+                url = String.format("jdbc:postgresql://%s:%d/", fields.getHost(), fields.getPort());
+                if (fields.getDatabase().isPresent()) {
+                    url += fields.getDatabase().get();
                 }
                 break;
             case "mysql":
                 // this includes mariadb
-                url = String.format("jdbc:mysql://%s:%d", host, port);
-                if (database.isPresent()) {
-                    url += "/" + database.get();
+                url = String.format("jdbc:mysql://%s:%d", fields.getHost(), fields.getPort());
+                if (fields.getDatabase().isPresent()) {
+                    url += "/" + fields.getDatabase().get();
                 }
                 break;
             case "oracle":
                 // using sid format, not service format
-                url = String.format("jdbc:oracle:thin:@%s:%d", host, port);
-                if (database.isPresent()) {
-                    url += ":" + database.get();
+                url = String.format("jdbc:oracle:thin:@%s:%d", fields.getHost(), fields.getPort());
+                if (fields.getDatabase().isPresent()) {
+                    url += ":" + fields.getDatabase().get();
                 }
                 break;
             default:
-                throw new UnsupportedOperationException("Don't know how to form a connection URL for JDBC driver " + vendorDriverId);
+                throw new UnsupportedOperationException("Don't know how to form a connection URL for JDBC driver " + fields.getVendorDriverId());
         }
         return url;
     }
+
+    public static class JdbcConnectionUrlFields {
+        private final String vendorDriverId;
+
+        private final String host;
+
+        private final int port;
+
+        private final Optional<String> database;
+
+        private final String hostAndPort;
+
+        public JdbcConnectionUrlFields(String vendorDriverId, String host, int port, Optional<String> database) {
+            this.vendorDriverId = requireNonNull(vendorDriverId);
```

Comment:
I'll add a message to the NPE thrown by requireNonNull(). Since this is a general constructor, parsing might not be happening while this object is being constructed, but, I'll add appropriate checks to parseJdbcConnectionUrl() to avoid passing nulls / empty values in.